src/Parser/Mage/Fire/Modules/Features/SearingTouch.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage, formatMilliseconds } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+
+const debug = false;
+
+class SearingTouch extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    abilityTracker: AbilityTracker,
+  };
+
+  badCasts = 0;
+  totalCasts = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.SEARING_TOUCH_TALENT.id);
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.FIREBALL.id || spellId === SPELLS.SCORCH.id) {
+      this.currentHealth = event.hitPoints;
+      this.maxHealth = event.maxHitPoints;
+    }
+    if (this.healthPercent < .30) {
+      this.totalCasts += 1;
```

Comment:
This includes all spells not just relevant spells, and so your util prop will be invalid.